sklearn/tree/_tree.pyx

```diff
@@ -920,9 +1110,36 @@ cdef inline void _init_split(SplitRecord* self, SIZE_t start_pos) nogil:
 
 
 cdef class Splitter:
+    """Abstract splitter class.
+
+    Splitters are called by tree builders to find the best splits on both
+    sparse and dense data, one split at a time.
+    """
+
     def __cinit__(self, Criterion criterion, SIZE_t max_features,
                   SIZE_t min_samples_leaf, double min_weight_leaf,
                   object random_state):
+        """
+        Parameters
+        ----------
+        criterion: Criterion
+            The criterion to measure the quality of a split. 
+
+        max_features: SIZE_t
+            The maximal number of features that a single decision tree can
+            consider.
```

Comment:
This is not correct. This is the maximal number of (randomly selected) features to consider for computing the next best split when growing the decision tree. If ``max_features == n_features`` all features will be considered to find the best split.

The full decision tree can make use of all the features for any value of `max_features`.